sihl-email/src/sihl_email.ml

```diff
@@ -64,61 +64,74 @@ let intercept sender email =
   else sender email
 ;;
 
-module Smtp : Sihl.Contract.Email.Sig = struct
-  include DevInbox
+module type SmtpConfig = sig
+  val sender : unit -> string Lwt.t
```

Comment:
@jerben I think this might not be a good approach. I think all these values should be in a record, because in case someone wants to read all these values from a file, they can't lazy-load it once, they have to load it for every single one of these functions. What do you think?

Reply:
I think it would make the API nicer, users probably don't want to use mutable state just to memoize it, they want to probably read a file and put it into a config in one go. Ok if I make an issue for it?